README.md

```diff
@@ -159,6 +159,41 @@ To get sample traces to try run the following command. The traces will be stored
 yarn download:sample-traces
 ```
 
+## Running UI tests
+
+To run the UI tests locally, use the following commands.
+
+Steps for setup that only need to be run once:
+
+```bash
+yarn download:sample-traces
+yarn download:server
+yarn download:openvscode-server
+yarn configure:openvscode-server
+yarn playwright install --with-deps
+```
+
+Steps to run once and again every time the application code is modified:
+
+```bash
+yarn
+yarn vsce:package
+# kill openvscode-server if running and restart it below
+```
+
+Steps to run once if the corresponding server is not already running
+
+```bash
+yarn start:server & # or run in a separate shell
+yarn start:openvscode-server & # or run in a separate shell
+```
+
+To run or re-run the tests after test code is modified:
+
+```bash
+yarn playwright test
+```
+
```

Comment:
Could you please add description on how to trouble-shoot failing tests? I think you can use something like this:
`yarn playwright show-trace test-results/extension-test-Open-Trace-from-Explorer-chromium/trace.zip`

also provide a line about retries, i.e.  `yarn playwright test --retries=3` will retry the tests